internal/exec/stages/disks/partitions.go

```diff
@@ -361,16 +498,32 @@ func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
 		matches := exists && matchErr == nil
 		wipeEntry := cutil.IsTrue(part.WipePartitionEntry)
 
+		var modification bool
+		var partInUse bool
+		for _, activePart := range activeParts {
+			prefix, err := partitionNumberPrefix(blockDevResolved)
+			if err != nil {
+				return err
+			}
+			if activePart == fmt.Sprintf("%s%s%d", blockDevResolved, prefix, part.Number) {
```

Comment:
Wonder if we should make it a map instead to avoid the nested loop.

Reply:
Found that there is `util.StrSliceContains` which I should use.